### DISCUSSION TITLE
Ignore a @JsonSubType in the JSON schema by custom annotation

### DISCUSSION THREAD

**Discussion by roded** [victools/jsonschema-generator#347] [Q&A] [answered]
Hi!,
I'd like to be able to control which `@JsonSubType`s are produced in the JSON schema by the victools jsonschema-generator.
This is useful when publishing JSON schemas but when some sub-types are for internal use.
Could this be somehow accomplished by annotating the sub-type with a custom annotation?  (Of course, it should not be `@JsonIgnoreType` since we do want these types to serialize).

```java
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME)
@JsonSubTypes({
    @JsonSubTypes.Type(value = InternalSubType.class, name = "internal-subtype"),
    @JsonSubTypes.Type(value = SubType.class, name = "subtype"),
})
public interface BaseType {

}

@Value
@HideFromJsonSchemaGenerator
public class InternalSubType implements BaseType {

    int field1;
    int field2;
}

@Value
public class SubType implements BaseType {

    int field1;
}
```

Thanks

**Comment by CarstenWickner** [comment, marked as answer]
HI @roded,

This is certainly possible, just not out-of-the-box.

I suggest you use the `JacksonModule` as per your current configuration but include the `JacksonOption.SKIP_SUBTYPE_LOOKUP` and `JacksonOption.IGNORE_TYPE_INFO_TRANSFORM`.
Then you apply its `JsonSubTypesResolver` yourself -- with a little override, e.g.
```java
JsonSubTypesResolver subtypeResolver = new JsonSubTypesResolver(jacksonOptions) {
    public List<ResolvedType> lookUpSubtypesFromAnnotation(ResolvedType declaredType, JsonSubTypes subtypesAnnotation, TypeContext context) {
        List<ResolvedType> subtypes = super.lookUpSubtypesFromAnnotation(declaredType, subtypesAnnotation, context);
        /* remove undesired subtypes from the list */
        return subtypes;
    }
};
generalConfigPart.withSubtypeResolver(subtypeResolver);
fieldConfigPart.withTargetTypeOverridesResolver(subtypeResolver::findTargetTypeOverrides);
methodConfigPart.withTargetTypeOverridesResolver(subtypeResolver::findTargetTypeOverrides);

generalConfigPart.withCustomDefinitionProvider(subtypeResolver);
fieldConfigPart.withCustomDefinitionProvider(subtypeResolver::provideCustomPropertySchemaDefinition);
methodConfigPart.withCustomDefinitionProvider(subtypeResolver::provideCustomPropertySchemaDefinition);
```

Unfortunately, I don't have the time to provide a full example in the next couple of days.
I hope the above gets you in the right direction already.

**Reply by roded** [reply]
Many thanks @CarstenWickner, I'll give it a shot.

**Reply by roded** [reply]
For my purpose, it seems like turning on `JacksonOption.SKIP_SUBTYPE_LOOKUP` and setting `configBuilder.forTypesInGeneral().withSubtypeResolver(...)` is enough.

Hopefully this change won't bite me later somehow. 

Thanks @CarstenWickner
